user_guide/configuration/suites.rst

```diff
@@ -10,71 +10,129 @@ With suites you can configure Behat to test different kinds of features
 using different kinds of contexts and doing so in one run. Test suites are
 really powerful and ``behat.yml`` makes them that much more powerful:
 
-.. code-block:: yaml
-
-    # behat.yml
-
-    default:
-        suites:
-            core_features:
-                paths:    [ '%paths.base%/features/core' ]
-                contexts: [ CoreDomainContext ]
-            user_features:
-                paths:    [ '%paths.base%/features/web' ]
-                filters:  { role: user }
-                contexts: [ UserContext ]
-            admin_features:
-                paths:    [ '%paths.base%/features/web' ]
-                filters:  { role: admin }
-                contexts: [ AdminContext ]
+.. code-block:: php
+
+    <?php
+    // behat.php
+
+    use App\Tests\Behat\Context\AdminContext;
+    use App\Tests\Behat\Context\CoreDomainContext;
+    use App\Tests\Behat\Context\UserContext;
+    use Behat\Config\Config;
+    use Behat\Config\Config\Filter\RoleFilter;
+    use Behat\Config\Profile;
+    use Behat\Config\Suite;
+
+    $profile = (new Profile('default'))
+        ->withSuite(
+            (new Suite('core_features'))
+                ->withPaths('%paths.base%/features/core')
+                ->withContexts(CoreDomainContext::class)
```

Comment:
On PHP8.4, we'll be able to skip the extra `()` around all the `new XXX()` which I think helps with readability. Maybe we should use 8.4 syntax in the docs with a comment the first time it appears, like so:

```suggestion
    // Note: on PHP < 8.4 you will need to use extra `()` around the `new` objects to allow method chaining
    // e.g.  `(new Profile('default'))`, `(new Suite('core_features'))`, etc.
    $profile = new Profile('default')
        ->withSuite(
             new Suite('core_features')
                ->withPaths('%paths.base%/features/core')
                ->withContexts(CoreDomainContext::class)
```

Reply:
Yes I agree.

Reply:
done